src/hooks/posters.ts

```diff
@@ -71,24 +71,53 @@ export const usePosters = (posterHallId: string) => {
   const fuseVenues = useMemo(
     () =>
       new Fuse(filteredPosterVenues, {
-        keys: ["poster.title", "poster.authorName", "poster.categories"],
+        keys: [
+          "name",
+          "poster.title",
+          "poster.authorName",
+          "poster.categories",
+        ],
+        threshold: 0.2, // 0.1 seems to be exact, default 0.6: brings too distant if anyhow related hits
+        ignoreLocation: true, // default False: True - to search ignoring location of the words.
+        findAllMatches: true,
       }),
     [filteredPosterVenues]
   );
 
   const searchedPosterVenues = useMemo(() => {
-    if (!searchQuery)
-      return filteredPosterVenues.slice(0, displayedPostersCount);
+    const normalizedSearchQuery = searchQuery.trim();
+
+    if (!normalizedSearchQuery) return filteredPosterVenues;
 
     return fuseVenues
-      .search(searchQuery)
-      .slice(0, displayedPostersCount)
+      .search({
+        //@ts-ignore
+        $and: normalizedSearchQuery
+          .match(/("[^"]*?"|[^"\s]+)+(?=\s*|\s*$)/g) // source: https://stackoverflow.com/a/16261693/1265472
```

Comment:
Should we extract this regex somewhere else, is there any chance it might be used in another place other than this one?

Reply:
do you have unit tests for the codebase? extra :+1: for moving away would then be -- would allow to get a nice unittest to ensure its correct operation.

Reply:
> do you have unit tests for the codebase?
Unfortunately, no

Reply:
Technically we have the infrastructure for it, but it needs a little love, and is only used in a couples of places at the moment (possible only one, testing the package-lock file). I believe the filename ends in .test.ts or similar if you wanted to find it.

Reply:
This was raised/shown on the previous PR but has been lost here since recreating as a new PR (which I have questions about in general @mike-lvov , as it complicates things in an I believe unnessecary way, let’s discuss on slack)

 ![D0BC2A4C-6C2F-4909-B626-2C6137846633](https://user-images.githubusercontent.com/753891/119898538-560a4000-bf85-11eb-9f76-bd2b5bdcbbc2.png)
![977D09BF-B995-4200-8126-09CF9036E797](https://user-images.githubusercontent.com/753891/119898555-5acef400-bf85-11eb-9300-011cf74620dd.png)

Reply:
This `!` issue remained a mystery since couldn't trigger and regex doesn't have any `!`. Concern is also IMHO minor since would only affect the malicious (or the one with stuck keyboard) user on the client side.

Reply:
<details>
<summary>looking at such a code analysis fiasco</summary> 

![image](https://user-images.githubusercontent.com/39889/119912090-bd45d700-bf28-11eb-8231-2adcdc225a4f.png)

where we can't get to that line if that variable is `None`
</details>

and the complexity of the regex analysis [codeql query](https://raw.githubusercontent.com/github/codeql/main/javascript/ql/src/Performance/ReDoS.ql) used for this one, I would definitely not get a sleepless night because of this "issue" and just consider it a "false positive" bug in the analysis query

Reply:
asked the "source": https://github.com/github/codeql/issues/5964

Reply:
Thanks for opening the issue about it @yarikoptic :)

Reply:
Looking at the 'explain' for the regex on https://regexr.com/ :

```
("[^"]*?"|[^"\s]+)+(?=\s*|\s*$)
```

<details><summary>Expand for screenshot..</summary>

![Screen Shot 2021-05-28 at 1 40 15 pm](https://user-images.githubusercontent.com/753891/119926209-435e2e00-bfba-11eb-9765-1a6833edd4ad.png)

</details>

Breaking this down:

- `(` `"[^"]*?"` `|` `[^"\s]+` `)+`
  - `"[^"]*?"`: Match `"`, not `"`, and then another `"`; 0 or more times, lazily
  - `|`: or
  - `[^"\s]+`: match not `"` or a space; 1 or more times
- `(?=` `\s*` `|` `\s*$` `)`
  - `(?=`: positive lookahead (matches a group after the main expression without including it in the result)
  - `\s*`: match 0 or more whitespace characters
  - `|`: or
  - `\s*$`: match 0 or more whitespace characters, then the end of the string
 
---
 
Won't look into this any further for now, and wait to see what (if anything) https://github.com/github/codeql/issues/5964 turns up.

Reply:
Fwiw, from what I am seeing in that explain, I don't think it's likely to be a false positive directly, though the `!` it mentions is likely incorrect (and possibly should refer to a `"` or a space based on the regex i'm guessing).

Can't quite wrap my 🧠 around it right now though to be able to definitively say that one way or another right now though.

Reply:
For 888 bytes, this lib seems to support it:

- https://github.com/jonschlinkert/split-string
  - seemingly not using regex: https://github.com/jonschlinkert/split-string/blob/master/index.js
- https://bundlephobia.com/result?p=split-string@6.1.0

Reply:
This is another SO that seems to provide some regex based solutions:

- https://stackoverflow.com/questions/2817646/javascript-split-string-on-space-or-on-quotes-to-array

Reply:
I like that split-words - has tests! Nice find.

Reply:
Re regex - ideally shouldn't split on - but if required, I guess could be improved. Didn't analyze/try yet.

Reply:
> ideally shouldn't split on `-` 

@yarikoptic Do you mean `_` (underscore, not dash)?

If so, I think the solution there is to just use `[a-zA-z0-9]` instead of `\w`

Reply:
In the interest of getting this merged, I'm going to call it OOS to muse on/change the regex/splitting method anymore here.

I've added a `@debt` comment to it in https://github.com/sparkletown/sparkle/pull/1460/commits/11cca352507b27f8d17583c7d0453347c590b920 referencing https://github.com/github/codeql/issues/5964 so that we have the relevant details captured there in future if needed.